cli/githubutil.py

```diff
@@ -71,6 +85,213 @@ def assign_github_team_to_repo(
+            queue.extend(not_visited_parents)
+            # queue.sort(key=lambda commit: commit.committed_date, reverse=True) #not needed anymore as we sort by semver at the end
+            visited |= set(_.map(not_visited_parents, lambda commit: commit.hexsha))
+
+    reachable_tags.sort(key=lambda t: parse_version_info(t), reverse=True)
+
+    if not reachable_tags:
+        warning('no release tag found, falling back to root commit')
+        root_commits = repo.iter_commits(rev=commit, max_parents=0)
+        root_commit = next(root_commits, None)
+        if not root_commit:
+            fail('could not determine root commit from rev {rev}'.format(rev=commit.hexsha))
+        if next(root_commits, None):
+            fail('cannot determine range for release notes. Repository has multiple root commits. Specify range via commit_range parameter.')
+        reachable_tags.append(root_commit.hexsha)
+
+    return reachable_tags
+
+
+def fetch_pr_numbers_in_range(
+    repo: git.Repo,
+    commit_range: str
+) -> set:
+    info('git log {range}'.format(range=commit_range))
+    gitLogs = repo.git.log(commit_range, pretty='%s').splitlines()
+    pr_numbers = []
+    for commitMessage in gitLogs:
+        if commitMessage.startswith('Merge pull'):
+            pr_number = _.head(re.findall(r"#(\d+|$)", commitMessage))
+            if pr_number:
+                pr_numbers.append(pr_number)
+
+    verbose('Merged pull request numbers in range {range}: {pr_numbers}'.format(range=commit_range, pr_numbers=pr_numbers))
+    return pr_numbers
+
+def fetch_release_notes_from_prs(
+    helper: GitHubRepositoryHelper,
+    pr_numbers_in_range: set
+) -> list:
+    # we should consider adding a release-note label to the PRs to reduce the number of search results
+    prs_iter = helper.search_issues_in_repo('type:pull is:closed')
+
+    release_notes = list()
+    for pr_iter in prs_iter:
+        pr_dict = pr_iter.as_dict()
+
+        pr_number = pr_dict['number']
+        if not str(pr_number) in pr_numbers_in_range:
+            continue
+
+        release_notes_pr = extract_release_notes(pr_number=pr_number, text=pr_dict['body'], user_login=_.get(pr_dict, 'user.login'))
+        if not release_notes_pr:
+            continue
+
+        release_notes.extend(release_notes_pr)
+    return release_notes
+
+def extract_release_notes(
+    pr_number: int,
+    text: str,
+    user_login: str
+) -> list:
+    release_notes = list()
+
+    quotes = re.findall(r"``` *(improvement|noteworthy)( (user|operator)?)?.*?\n(.*?)\n```", text, re.MULTILINE | re.DOTALL)
+    for quote in quotes:
+        quote = _.map(quote, lambda obj: _.trim(obj))
+
+        text = quote[3]
+        if not text or 'none' == text.lower():
+            continue
+
+        category = quote[0]
+        group = quote[2] or 'user'
+
+        release_notes.append(ReleaseNote(category_id=category, group_id=group, text=text, pr_number=pr_number, user_login=user_login))
+    return release_notes
```

Comment:
yield

Reply:
see comment from above (I have to filter on the list)